src/core/metadata/mod.rs

```diff
@@ -47,6 +49,43 @@ impl WalletMetadata {
         &mut self.2
     }
 
+    /// Return a mutable reference to the inner untyped object.
+    /// This is useful for adding additional parameters to the wallet metadata,
+    /// such as `client_id_schemes_supported`.
+    ///
+    /// NOTE: There is a typed method for adding a `client_id_schemes_supported` parameter.
+    /// See:
+    pub fn untyped_object_mut(&mut self) -> &mut UntypedObject {
+        &mut self.0
+    }
+
+    /// Add a client ID scheme to the list of the client ID schemes supported.
+    ///
+    /// This method will construct a `client_id_schemes_supported` proprety in the
+    /// wallet metadata if none exists previously, otherwise, this method will add
+    /// the client ID scheme to the existing list of the client ID schemes supported.
+    pub fn add_client_id_schemes_supported(&mut self, client_id_scheme: ClientIdScheme) {
+        let mut supported = match self.0.remove::<ClientIdSchemesSupported>() {
+            Some(Ok(supported)) => supported,
+            Some(Err(e)) => {
+                // NOTE: Not handling an error here, but warning if the client id schemes
+                // cannot be parsed. We may wish to handle this error in the future.
+                tracing::warn!(
+                    "Failed to parsed client id schemes supported from wallet metadata: {e:?}"
+                );
+
+                ClientIdSchemesSupported::default()
+            }
+            _ => ClientIdSchemesSupported::default(),
+        };
+
+        // Insert the scheme.
+        supported.0.push(client_id_scheme);
+
+        // Insert the updated client IDs schemes supported.
+        self.0.insert(supported);
+    }
```

Comment:
I would prefer to not add this to the library. It's easy enough for an implementer to do `metadata.deref_mut()...` to access the UntypedObject and do this themselves. If we add this function then it begs the question of why don't we support setting, updating and mutating for all metadata parameters.

Reply:
I think it's okay to add methods as needed.